### PR TITLE
Combine Op

### LINKED ISSUE #1047: Update Combine operator

### PR DESCRIPTION
**Pull request by KateUnger**
**Description of changes:**
- Update Combine 

TODO: Change `cast_kernels.cpp` and c`ast_kernels.cu`

**Related Issues:**

Linked Issues:
- Issue #1047

Issues closed by this PR:
- Closes #1047

- - -
This change is [Reviewable](https://reviewable.io/reviews/flexflow/FlexFlow/1113)

### CODE REVIEW

File: lib/runtime/src/ops/combine.cc [hunk 1]
```diff
@@ -13,322 +13,112 @@
...
 
 using namespace FlexFlow::Kernels::Combine;
 
-CombineParams Combine::get_params() const {
-  CombineParams params;
-  params.combine_legion_dim = this->combine_dim;
-  params.combine_degree = this->combine_degree;
-  return params;
-}
+enum Slots { INPUT, OUTPUT, PROFILING };
```
Comment (reyna-abhyankar) [inline lib/runtime/src/ops/combine.cc:29]: Why not `ATTRS`?

Comment (KateUnger) [reply lib/runtime/src/ops/combine.cc:29]: I don't use any of the attrs properties, so I don't need to get or bind them (I think)

File: lib/runtime/src/ops/combine.cc [hunk 2]
```diff
@@ -13,322 +13,131 @@
...
-    Runtime *runtime) {
-  Domain output_grad_domain = runtime->get_index_space_domain(
-      ctx, task->regions[0].region.get_index_space());
-  Domain input_grad_domain = runtime->get_index_space_domain(
-      ctx, task->regions[1].region.get_index_space());
-  assert(output_grad_domain == input_grad_domain);
+template <>
+OpTaskSignature bwd_signature<COMBINE_BWD_TASK_ID>() {
+  OpTaskSignature bwd =
+      infer_bwd_signature(get_op_signature(COMBINE_FWD_TASK_ID));
```
Comment (reyna-abhyankar) [inline lib/runtime/src/ops/combine.cc:128]:
```suggestion
      infer_bwd_signature(fwd_signature<COMBINE_FWD_TASK_ID>());
```

File: lib/runtime/src/ops/combine.cc [hunk 3]
```diff
@@ -13,322 +13,131 @@
...
-namespace std {
-size_t hash<FlexFlow::CombineParams>::operator()(
-    FlexFlow::CombineParams const &params) const {
-  size_t key = 0;
-  hash_combine(key, params.combine_legion_dim);
-  hash_combine(key, params.combine_degree);
-  return key;
-}
-}; // namespace std
+namespace std {}; // namespace std
```
Comment (reyna-abhyankar) [inline lib/runtime/src/ops/combine.cc:143]:
```suggestion
```

**Review by reyna-abhyankar — DISMISSED** [review]

**Review by reyna-abhyankar — APPROVED** [review]

**Pull request merged**